protocol/testing/e2e/gov/add_new_market_test.go

```diff
@@ -0,0 +1,256 @@
+		proposedMsgs             []sdk.Msg
+		updateClobDelayBlocks    uint32
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success with 4 standard messages": {
+			proposedMsgs: []sdk.Msg{
+				&pricestypes.MsgCreateOracleMarket{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testMarketParam1001.Param,
+				},
+				&perptypes.MsgCreatePerpetual{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testPerpetual1001.Params,
+				},
+				&clobtypes.MsgCreateClobPair{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					ClobPair:  *testClobPair1001,
+				},
+				&delaymsgtypes.MsgDelayMessage{
+					Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive),
+					DelayBlocks: 10,
+				},
+			},
+			updateClobDelayBlocks:  10,
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Proposals execution fails due to incorrectly ordered messages": {
+			proposedMsgs: []sdk.Msg{
+				&pricestypes.MsgCreateOracleMarket{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testMarketParam1001.Param,
+				},
+				// Create clob pair before creating perpetual, which will fail.
+				&clobtypes.MsgCreateClobPair{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					ClobPair:  *testClobPair1001,
+				},
+				&perptypes.MsgCreatePerpetual{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params:    testPerpetual1001.Params,
+				},
+				&delaymsgtypes.MsgDelayMessage{
+					Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive),
+					DelayBlocks: 10,
+				},
+			},
+			updateClobDelayBlocks:  10,
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+		},
+		"Proposal execution fails due to existing objects": {
+			proposedMsgs: []sdk.Msg{
+				&pricestypes.MsgCreateOracleMarket{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params: pricestest.GenerateMarketParamPrice(
+						pricestest.WithId(5),
+					).Param,
+				},
+				&perptypes.MsgCreatePerpetual{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params: perptest.GeneratePerpetual(
+						perptest.WithId(5),
+					).Params,
+				},
+				&clobtypes.MsgCreateClobPair{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					ClobPair: *clobtest.GenerateClobPair(
+						clobtest.WithId(5),
+					),
+				},
+				&delaymsgtypes.MsgDelayMessage{
+					Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive),
```

Comment:
Should we add a test case where the delay msg is incorrect so although the gov passed, the CLOB market never gets updated to ACTIVE?

Reply:
Added